platform/machine/aws/machine.go

```diff
@@ -113,7 +113,12 @@ func (am *machine) saveConsole(origConsole string) error {
 		return true, nil
 	})
 	if err != nil {
-		return fmt.Errorf("retrieving console output of %v: %v", am.ID(), err)
+		err = fmt.Errorf("retrieving console output of %v: %v", am.ID(), err)
+		if origConsole != "" {
+			plog.Warning(err)
```

Comment:
I just realized: at this point `am.console` may equal `origConsole` or may be empty, and we're relying on the matching heuristic below to ensure that they're correctly merged.  It might be a good idea to add special cases below to ensure this is handled correctly.

Reply:
https://play.golang.org/p/La1BAPAVT2O

It looks like it still works with these cases, was there any others that you had in mind?

Reply:
The second case was a thinko on my part; `LastIndex` will return the index of the end of the string, so we won't lose any of `origConsole`.  As to the first case, the heuristic may be confused if the first 100 chars of `am.console` occur more than once in `origConsole` (e.g. if there's log spam).  That's true anyway, but in this special case we have additional information we can use to avoid that confusion, namely that the two strings are identical.  I don't think it's a dealbreaker though.